src/error.rs

```diff
@@ -10,11 +11,21 @@ pub enum Error {
     Message(MessageError),
     /// The system IO returned an error.
     Io(IoError),
+    /// Generic error when attempting to parse a url
+    // TODO: Some of the specific errors should be named, maybe a separate enum
+    // for the errors encountered during url parsing should be used
+    UrlParsing(Box<StdError + Send + Sync>),
```

Comment:
Personal Preference: I'd call this just `Url`. Also it would be nice to have an explicit UrlError type to go in this variant.